call_fd_callback.c

```diff
@@ -0,0 +1,12 @@
+#include <stdint.h>
+
+typedef void (*fd_callback)(void *class_ptr, int);
+
+// bridge to allow a function pointer to be passed to go. That ptr must be called from c.
+void call_fd_callback(void* func_ptr, void* class_ptr, uintptr_t fd)
```

Comment:
Go cannot call the function pointer directly. We must pass it to a C function which then calls it. This is compiled into the library itself - no build changes are needed.